h5grove/models.py

```diff
@@ -1,11 +1,12 @@
+from __future__ import annotations
 from enum import Enum
-from typing import Dict, Tuple, Union
-from typing_extensions import TypedDict
+from typing import Union, Tuple, Dict
+from typing_extensions import TypedDict, NotRequired
 import h5py
 
-H5pyEntity = Union[
-    h5py.Dataset, h5py.Datatype, h5py.ExternalLink, h5py.Group, h5py.SoftLink
-]
+H5pyEntity = (
+    h5py.Dataset | h5py.Datatype | h5py.ExternalLink | h5py.Group | h5py.SoftLink
+)
 
 Selection = Union[str, int, slice, Tuple[Union[slice, int], ...], None]
```

Comment:
I couldn't find a way to replace all the `Union` and `Tuple` in this file. Kept getting errors. Not sure if you want to have a go?

Reply:
Okay, so I guess it's normal.

Reply:
It's "normal" -> https://github.com/silx-kit/h5grove/pull/97#discussion_r1735837896
It's actually possible with recent versions of Python.

Reply:
Seems there is no way around it: https://github.com/silx-kit/h5grove/pull/97#discussion_r1735837896